src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1355,6 +1355,28 @@ public static bool IsPartial(this TypeSymbol type)
             return type is SourceNamedTypeSymbol { IsPartial: true };
         }
 
+        public static bool IsFileTypeOrUsesFileTypes(this TypeSymbol type)
+        {
+            var foundType = type.VisitType(predicate: (type, _, _) => type is SourceMemberContainerTypeSymbol { IsFile: true }, arg: (object?)null);
+            return foundType is not null;
+        }
+
+        public static bool IsFileTypeInSeparateFileFrom(this SourceNamedTypeSymbol possibleFileType, SourceNamedTypeSymbol otherSymbol)
```

Comment:
nit: do we think this will get re-used? If not consider making this a local function at single call-site

Reply:
I thought this might get reused in another place where we report redundant symbols, but it didn't end up being the case. Moving to local function.